### ISSUE TITLE
Chrome android: keyboard focus in Callout causes scroll from keyboard, which dismisses Callouts

### ISSUE THREAD

**Issue by pablonete** [microsoft/fluentui#6864]
### Codepen

https://codepen.io/pablonete/pen/zmeBRR?editors=0010 

### Bug Report 

We have a dropdown list with a filter input inside the callout. When used in Chrome for Android, clicking on the TextField makes the on-screen keyboard to show up, which in turn dismisses the callout. So the dropdown is unable to use.
- __Package version(s)__: Latest
- __Browser and OS versions__: Chrome on Android
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? I cannot figure out any the solution yet :/
 
Requested priority: Low
 
Products/sites affected: Azure DevOps

**Comment by cliffkoh** [comment]
I was unable to repro on devices running Android 7 and Android 8 (I believe it was a Motorola and a Samsung Galaxy).

Could you please test on a few other devices and provide some information on how widespread you think this problem is?

**Comment by pablonete** [comment]
Thank you @cliffkoh. Discussed with @micahgodbolt it seems to be caused by custom Android keyboards. I confirmed the bug in my Huaway P8 with both SwiftKey and GBoard. I couldn't find a way to test with the built-in keyboard.

**Comment by vbjay** [comment]
https://developercommunity.visualstudio.com/content/problem/341264/mobile-repository-dropdown.html. 

I raised this issue and the codepen does not exhibit the issue in that issue. I can try altering it to make it happen.

**Comment by cliffkoh** [comment]
Hi @pablonete, like @vbjay, I am unable to repro on with the SwiftKey keyboard either.

**Comment by pablonete** [comment]
I have a consistent repro on my Huawai P8 Lite with custom layer EMUI 8.0 on Android 8.0. I click on Show callout, then when I click on any of the 2 inputs, callout quickly dismisses. These are the 2 events that I'm seeing in quick sequence, both firing through onDismiss, see updated codepen.

```
1. {
 ".ctor": "Event",
 "isTrusted": true,
 "NONE": 0,
 "CAPTURING_PHASE": 1,
 "AT_TARGET": 2,
 "BUBBLING_PHASE": 3,
 "type": "resize",
 "target": "<Window>",
 "currentTarget": "<Window>",
 "eventPhase": 2,
 "bubbles": false,
 "cancelable": false,
 "defaultPrevented": false,
 "composed": false,
 "timeStamp": 2825.299999996787,
 "srcElement": "<Window>",
 "returnValue": true,
 "cancelBubble": false,
 "path": [
  "<Window>"
 ]
}
2. {
 ".ctor": "FocusEvent",
 "isTrusted": true,
 "relatedTarget": "<input type=\"text\" id=\"TextFie",
 "view": "<Window>",
 "detail": 0,
 "sourceCapabilities": null,
 "which": 0,
 "NONE": 0,
 "CAPTURING_PHASE": 1,
 "AT_TARGET": 2,
 "BUBBLING_PHASE": 3,
 "type": "focus",
 "target": "<button type=\"button\" class=\"m",
 "currentTarget": null,
 "eventPhase": 0,
 "bubbles": false,
 "cancelable": false,
 "defaultPrevented": false,
 "composed": true,
 "timeStamp": 2867.899999997462,
 "srcElement": "<button type=\"button\" class=\"m",
 "returnValue": true,
 "cancelBubble": false,
 "path": [
  "<button type=\"button\" class=\"m",
  "<div class=\"ms-Fabric foo root",
  "<div id=\"content\"><div class=\"",
  "<body translate=\"no\">\n\n  <scri",
  "<html lang=\"en\" dir=\"ltr\"><hea",
  null,
  "<Window>"
 ]
}
```

It seems we have a window resize, and then focus is set to the page button. Does it make sense?

**Comment by vbjay** [comment]
When I click button on codepen and tap text field keyboard shows and
drop-down does not hide. I have to tap in page to hide the keyboard. This
is what I see in the log when I do so.

The difference I see is on the devops site the keyboard shows as soon as I
drop down the repo select.

Events
{
".ctor": "MouseEvent",
"isTrusted": true,
"screenX": 255,
"screenY": 451,
"clientX": 510,
"clientY": 33,
"ctrlKey": false,
"shiftKey": false,
"altKey": false,
"metaKey": false,
"button": 0,
"buttons": 0,
"relatedTarget": null,
"pageX": 510,
"pageY": 33,
"x": 510,
"y": 33,
"offsetX": 502,
"offsetY": 16,
"movementX": 0,
"movementY": 0,
"fromElement": null,
"toElement": "<h2>Try</h2>",
"layerX": 510,
"layerY": 33,
"view": "<Window>",
"detail": 1,
"sourceCapabilities": {},
"which": 1,
"NONE": 0,
"CAPTURING_PHASE": 1,
"AT_TARGET": 2,
"BUBBLING_PHASE": 3,
"type": "click",
"target": "<h2>Try</h2>",
"currentTarget": "<html lang=\"en\" class=\"\" dir=\"ltr\"><head",
"eventPhase": 1,
"bubbles": true,
"cancelable": true,
"defaultPrevented": false,
"composed": true,
"timeStamp": 7100.399999995716,
"srcElement": "<h2>Try</h2>",
"returnValue": true,
"cancelBubble": false,
"path": [
"<h2>Try</h2>",
"<div class=\"ms-Fabric foo root-47\"><h2>T",
"<div id=\"content\"><div class=\"ms-Fabric ",
"<body>\n<script src=\"//unpkg.com/react@16",
"<html lang=\"en\" class=\"\" dir=\"ltr\"><head",
null,
"<Window>"
]
}
{
".ctor": "FocusEvent",
"isTrusted": true,
"relatedTarget": null,
"view": "<Window>",
"detail": 0,
"sourceCapabilities": null,
"which": 0,
"NONE": 0,
"CAPTURING_PHASE": 1,
"AT_TARGET": 2,
"BUBBLING_PHASE": 3,
"type": "focus",
"target": "<button type=\"button\" class=\"ms-Button m",
"currentTarget": null,
"eventPhase": 0,
"bubbles": false,
"cancelable": false,
"defaultPrevented": false,
"composed": true,
"timeStamp": 7164.4000000014785,
"srcElement": "<button type=\"button\" class=\"ms-Button m",
"returnValue": true,
"cancelBubble": false,
"path": [
"<button type=\"button\" class=\"ms-Button m",
"<div class=\"ms-Fabric foo root-47\"><h2>T",
"<div id=\"content\"><div class=\"ms-Fabric ",
"<body>\n<script src=\"//unpkg.com/react@16",
"<html lang=\"en\" class=\"\" dir=\"ltr\"><head",
null,
"<Window>"
]
}

**Comment by pablonete** [comment]
Thanks @vbjay. You may want to try autofocus={true} on TextField or input, I left it out because it's not needed for my minimal repro.
It seems clear to me that showing the keyboard is resizing the viewport in some cases, and this resize event hides the popup.

**Comment by dzearing** [comment]
Yeah this makes sense to me; the callout will dismiss on scroll unless you tell it not to. You can certainly tell it to avoid dismissal using `preventDismissOnScroll` prop.

**Comment by dzearing** [comment]
I'd clearly love to just do the right thing by default here. Do you have suggestions?

There isn't a way to differentiate between a keyboard soft scroll and a normal scroll, afaik. https://stackoverflow.com/questions/27132796/is-there-any-javascript-event-fired-when-the-on-screen-keyboard-on-mobile-safari

Just a rough idea. If focus is placed within a layer and a scroll occurs in a short timeframe, we don't dismiss. If they scroll (not due to focus shift within the callout) we still dismiss.

**Comment by dzearing** [comment]
Marking as a bug, as I don't feel like this is the ideal. Any time a callout appears, anything which shifts scroll will dismiss it, when really just adjusting position would be good enough.

**Comment by cliffkoh** [comment]
@dzearing should this be added to shield board too?

**Comment by dzearing** [comment]
@cliffkoh I think we want to retire the shield board. I think we need to just distribute bugs to area owners but shield could do some experiments.

I'd like to at least have a clear goal here though first. We could certainly experiment with avoiding the dismissal on "scroll from focus of element within callout". I'm not sure if this will cause additional issues, so I think the goal would be see if that clears the concern without regressing normal use cases.
